src/Moryx.Launcher/ShellNavigator.cs

```diff
@@ -72,32 +73,51 @@ public async Task<IReadOnlyList<WebModuleItem>> GetWebModuleItems(HttpContext co
                 }).ToArray();
             }
 
-            var webModules = compiledPageActionDescriptors.Select(cpad => CreateWebModuleItem(cpad))
-                .OfType<WebModuleItem>().ToList();
+            // Load modules
+            var modules = compiledPageActionDescriptors.Select(CreateWebModuleItem)
+                .Where(m => m != null).ToList<ModuleItem>();
+
+            // Load external modules
+            var externalModuleConfigs = _configuration.GetSection("Shell:ExternalModules").Get<ExternalModuleConfig[]>();
```

Comment:
Do you want to split this into a separate method and reuse it in the cshtml of the page? Then we don't have the configuration string in two places

Reply:
And do we need to update the config during runtime, or should we get it once during startup and store it for the runtime? 🤔

Reply:
Adding to the appsettings and refresh the page is nice but I also affects the loading time of the whole application. Will move it to one-time load.